session/session.go

```diff
@@ -0,0 +1,113 @@
+package session
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"launchpad.net/gozk"
+)
+
+var (
+	ErrZkSessionNotConnected       = fmt.Errorf("Unable to connect to ZooKeeper.")
+	ErrZkSessionTimeout            = fmt.Errorf("Session connection timeout expired.")
+	ErrZkSessionExpired            = fmt.Errorf("Session expired.")
+	ErrZkSessionEventChannelClosed = fmt.Errorf("Session event channel closed.")
+)
+
+type ZkSession struct {
+	servers     string
+	dialTimeout time.Duration
+	recvTimeout time.Duration
+	Connection  *zookeeper.Conn
+	events      <-chan zookeeper.Event
+	mu          sync.Mutex
+}
+
+func NewZkSession(servers string, dialTimeout time.Duration, recvTimeout time.Duration) (*ZkSession, error) {
+	zkconn, zkevents, err := zookeeper.Dial(servers, recvTimeout)
+
+	if err != nil {
```

Comment:
Total micro-nitpick that you can feel free to ignore, but I like to not leave a blank space between the action and the error handler, so it's 4 lines all together rather than 2 paragraphs.